File: your-financeje-banking/src/test/java/org/jesperancinha/fintech/model/AccountTest.java
package org.jesperancinha.fintech.model;

import junit.framework.TestCase;
import lombok.val;

import java.math.BigDecimal;

import static org.assertj.core.api.Assertions.assertThat;

public class AccountTest extends TestCase {

    public void testAddCurrentValue() {
        val account = Account.builder().accountNumber("SDFSFS234862398643398246345345345928").build();
        val accountNew = account.addCurrentValue(100L);

        assertThat(accountNew.accountNumber()).isEqualTo("SDFSFS234862398643398246345345345928");
        assertThat(accountNew.creditValue()).isEqualTo(BigDecimal.valueOf(0L));
        assertThat(accountNew.currentValue()).isEqualTo(BigDecimal.valueOf(100L));
    }

    public void testAddCreditValue() {
        val account = Account.builder().accountNumber("SDFSFS234862398643398246345345345928").build();
        val accountNew = account.addCreditValue(10L);

        assertThat(accountNew.accountNumber()).isEqualTo("SDFSFS234862398643398246345345345928");
        assertThat(accountNew.creditValue()).isEqualTo(BigDecimal.valueOf(10L));
        assertThat(accountNew.currentValue()).isEqualTo(BigDecimal.valueOf(0L));
    }
}

File: your-financeje-banking/src/test/kotlin/org/jesperancinha/fintech/model/AccountTest.kt
package org.jesperancinha.fintech.model

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import java.math.BigDecimal

class AccountTest {
    @Test
    fun `should add current value`() {
        val account = Account(
            accountNumber = "SDFSFS234862398643398246345345345928"
        )
        val accountNew = account.addCurrentValue(100L)
        accountNew.accountNumber shouldBe "SDFSFS234862398643398246345345345928"
        accountNew.creditValue shouldBe BigDecimal.valueOf(0L)
        accountNew.currentValue shouldBe BigDecimal.valueOf(100L)
    }

    @Test
    fun `should add credit value`() {
        val account = Account(
            accountNumber = "SDFSFS234862398643398246345345345928"
        )
        val accountNew = account.addCreditValue(10L)
        accountNew.accountNumber shouldBe "SDFSFS234862398643398246345345345928"
        accountNew.creditValue shouldBe BigDecimal.valueOf(10L)
        accountNew.currentValue shouldBe BigDecimal.valueOf(0L)
    }
}